AU2/plugins/custom_plugins/WantedPlugin.py

```diff
@@ -102,130 +117,122 @@ def on_event_update(self, e: Event, htmlResponse) -> List[HTMLComponent]:
+        dead_players_times = []
+        dead_police_times = []
+        # Support for police dying multiple times
+        # Intentionally has multiple copies of police if they die while corrupt multiple times
+        for id in set(death_manager.get_dead()):
+            player = ASSASSINS_DATABASE.get(id)
+            if player.is_police:
+                for time in death_manager.death_timings[id]:
+                    if wanted_manger.is_wanted_at(player, time):
+                        dead_police_times.append((player, time))
+            else:  # Assume regular player's first death is only relevant one
+                if wanted_manger.is_wanted_at(player, death_manager.death_timings[id][0]):
+                    dead_players_times.append((player, death_manager.death_timings[id][0]))
+
+        tables = []
+        if wanted_players:
+            rows = []
+            for player in wanted_players:
+                (_, crime, redemption) = wanted_manger.get_wanted_player_crime_info(player, now)[1]
+                rows.append(
+                    PLAYER_TABLE_ROW_TEMPLATE.format(
+                        REAL_NAME=escape(player.real_name),
+                        PSEUDONYMS=escape(player.all_pseudonyms()),
+                        ADDRESS=escape(player.address),
+                        COLLEGE=escape(player.college),
+                        WATER_STATUS=escape(player.water_status),
+                        CRIME=escape(crime),
+                        REDEMPTION=escape(redemption),
+                        NOTES=escape(player.notes)
+                    )
                 )
+            tables.append(
+                PLAYER_TABLE_TEMPLATE.format(ROWS="".join(rows))
             )
-
-        for playerID in wanted_police:
-            player = ASSASSINS_DATABASE.get(playerID)
-            (_, crime, redemption) = wanted_police[playerID]
-            police_rows.append(
-                POLICE_TABLE_ROW_TEMPLATE.format(
-                    RANK="Police", # TODO: update when police rank plugin implemented
-                    REAL_NAME=escape(player.real_name),
-                    PSEUDONYMS=escape(player.all_pseudonyms()),
-                    ADDRESS=escape(player.address),
-                    COLLEGE=escape(player.college),
-                    WATER_STATUS=escape(player.water_status),
-                    CRIME=escape(crime),
-                    REDEMPTION=escape(redemption),
-                    NOTES=escape(player.notes)
+        if wanted_police:
+            rows = []
+            for player in wanted_police:
+                (_, crime, redemption) = wanted_manger.get_wanted_player_crime_info(player, now)[1]
+                rows.append(
+                    POLICE_TABLE_ROW_TEMPLATE.format(
+                        RANK="Police",
+                        # TODO: update when police rank plugin refactored such that it is possible to get the rank name,

```

Comment:
There calls for a police rank manager. You could mention that in the to-do so that whoever picks this up knows where to start